accumulo-store/src/test/java/gaffer/accumulostore/key/impl/RowIdAggregatorTest.java

```diff
@@ -0,0 +1,274 @@
+
+            // Write mutation
+            final BatchWriterConfig writerConfig = new BatchWriterConfig();
+            writerConfig.setMaxMemory(1000000L);
+            writerConfig.setMaxLatency(1000L, TimeUnit.MILLISECONDS);
+            writerConfig.setMaxWriteThreads(1);
+            final BatchWriter writer = store.getMockConnector().createBatchWriter(store.getProperties().getTable(), writerConfig);
+            writer.addMutation(m1);
+            writer.addMutation(m2);
+            writer.addMutation(m3);
+            writer.addMutation(m4);
+            writer.addMutation(m5);
+            writer.addMutation(m6);
+            writer.addMutation(m7);
+            writer.addMutation(m8);
+            writer.addMutation(m9);
+            writer.close();
+
+            // Read data back and check we get one merged element
+            final Authorizations authorizations = new Authorizations(visibilityString);
+            final BatchScanner scanner = store.getMockConnector().createBatchScanner(store.getProperties().getTable(), authorizations, 1000);
+            try {
+                scanner.addScanIterator(store.getKeyPackage().getIteratorFactory().getRowIDAggregatorIteratorSetting(store, "BasicEdge2"));
+            } catch (IteratorSettingException e) {
+                fail(e.getMessage());
+            }
+            RangeFactory rangeF = store.getKeyPackage().getRangeFactory();
+            Range r = rangeF.getRangeFromPair(new Pair<ElementSeed>((new EntitySeed("1")), new EntitySeed("4")), new SummariseGroupOverRanges());
+            Range r2 = rangeF.getRangeFromPair(new Pair<ElementSeed>((new EntitySeed("5")), new EntitySeed("5")), new SummariseGroupOverRanges());
+            scanner.setRanges(Arrays.asList(r, r2));
+            final Iterator<Map.Entry<Key, Value>> it = scanner.iterator();
+            Map.Entry<Key, Value> entry = it.next();
+            Element readEdge = elementConverter.getFullElement(entry.getKey(), entry.getValue());
+
+            Edge expectedEdge = new Edge("BasicEdge2");
+            expectedEdge.setSource("4");
+            expectedEdge.setDestination("8");
+            expectedEdge.setDirected(true);
+            expectedEdge.putProperty(AccumuloPropertyNames.COLUMN_QUALIFIER, 5);
+            expectedEdge.putProperty(AccumuloPropertyNames.PROP_1, 0);
+            expectedEdge.putProperty(AccumuloPropertyNames.PROP_2, 0);
+            expectedEdge.putProperty(AccumuloPropertyNames.PROP_3, 0);
+            expectedEdge.putProperty(AccumuloPropertyNames.PROP_4, 0);
+            expectedEdge.putProperty(AccumuloPropertyNames.COUNT, 3);
+
+            assertEquals(expectedEdge, readEdge);
+            assertEquals(5, readEdge.getProperty(AccumuloPropertyNames.COLUMN_QUALIFIER));
+            assertEquals(3, readEdge.getProperty(AccumuloPropertyNames.COUNT));
+            // Check no more entries
```

Comment:
This comment doesn't make sense - the next line asserts that more entries do exist.